domain/camera/src/main/java/com/google/jetpackcamera/domain/camera/CameraUseCase.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.jetpackcamera.domain.camera
+
+import androidx.camera.core.CameraSelector
+import androidx.camera.core.Preview
+import androidx.lifecycle.LifecycleOwner
+
+/**
+ * Data layer for camera.
+ */
+interface CameraUseCase {
+
+    /**
+     * Initializes the camera.
+     */
+    suspend fun initialize(): List<Int>
+
+    /**
+     * Starts the camera preview.
+     */
+    fun startPreview(
+        lifecycleOwner: LifecycleOwner,
+        surfaceProvider: Preview.SurfaceProvider,
+        @CameraSelector.LensFacing lensFacing: Int
```

Comment:
Since the intent of this interface may be to provide an abstraction layer to switch b/w Camera2 and CameraX it'll be hard to do so with this interface since these are more CameraX specific deps.
We don't have to change it right now but just an FYI

Reply:
That's a good catch. I'll change it to be more compatible.